### PR TITLE
DDC -> dartdevc, add links to dartdevc docs

### PR DESCRIPTION
**Pull request by kwalrath**
Staged at:

https://kw-www-dartlang-1.firebaseapp.com/guides/language/analysis-options
https://kw-www-dartlang-1.firebaseapp.com/guides/language/effective-dart/usage
https://kw-www-dartlang-1.firebaseapp.com/guides/language/sound-dart
https://kw-www-dartlang-1.firebaseapp.com/guides/language/sound-faq
https://kw-www-dartlang-1.firebaseapp.com/guides/language/sound-problems
https://kw-www-dartlang-1.firebaseapp.com/tools/analyzer

### CODE REVIEW

File: src/_guides/language/sound-dart.md [hunk 1]
```diff
@@ -736,15 +737,16 @@ strong mode:
...
 but most of the information applies to anyone using strong mode Dart:
 
 * [Strong Mode](https://github.com/dart-lang/dev_compiler/blob/master/STRONG_MODE.md) -
   Motivation for strong mode Dart.
 * [Strong Mode Static Checking](https://github.com/dart-lang/dev_compiler/blob/master/doc/STATIC_SAFETY.md) -
   Type inference in strong mode Dart.
 * [Strong Mode in the Dart Dev Compiler](https://chromium.googlesource.com/external/github.com/dart-lang/dev_compiler/+/refs/heads/master/doc/RUNTIME_SAFETY.md) -
-  Runtime checks in DDC.
+  Runtime checks in dartdevc.
 * [Prototype Syntax for Generic Methods](https://github.com/dart-lang/dev_compiler/blob/master/doc/GENERIC_METHODS.md) -
```
Comment (vsmenon) [inline src/_guides/language/sound-dart.md]: No longer prototype - we now have real generic methods.

File: src/_guides/language/sound-problems.md [hunk 2]
```diff
@@ -240,8 +240,8 @@ For more information, see [Use proper input parameter types when overriding meth
 **Warning:**
 <code>Unsound implicit cast from <em>Class&lt;dynamic&gt;</em> to <em>Class&lt;type&gt;</em>.</code>
```
Comment (vsmenon) [inline src/_guides/language/sound-problems.md]: This is no longer a warning.

File: src/_guides/language/sound-problems.md [hunk 3]
```diff
@@ -455,7 +456,7 @@ in [Effective Dart](/guides/language/effective-dart/).
 
 Strong mode currently disallows overriding a field with another field or with a getter/setter. Generally speaking making the superclass field into a getter/setter pair is a simple solution.  Note that we do allow an interface which declares a field to be implemented with a field, or with a getter/setter. Overriding a field with another field or with a getter/setter pair generally leaves the old field around using up memory in the object. It’s technically still possible to access the field, but most code simply leaves it around unused.
```
Comment (vsmenon) [inline src/_guides/language/sound-problems.md]: Is this in a comment?  Both strong mode and DDC now allow/support field overrides.

Comment (kwalrath) [inline src/_guides/language/sound-problems.md]: Yes that's in a comment. I'll delete it.

File: src/_guides/language/sound-problems.md (hunk 2 above)
Comment (kwalrath) [inline src/_guides/language/sound-problems.md]: Do you mean that it's an error, or that that message no longer appears?

If the latter, should I rewrite the https://kw-www-dartlang-1.firebaseapp.com/guides/language/sound-problems#missing-type-arguments section to talk only about the "Invalid override" error? Do I need to change the example or any of the other text?

File: src/_guides/language/sound-dart.md (hunk 1 above)
Comment (kwalrath) [inline src/_guides/language/sound-dart.md]: OK, I'll rename that link. I should probably link (earlier) to https://www.dartlang.org/guides/language/language-tour#using-generic-methods

File: src/_guides/language/sound-problems.md (hunk 2 above)
Comment (vsmenon) [inline src/_guides/language/sound-problems.md]: Sorry, I meant that it no longer appears.

Comment (vsmenon) [inline src/_guides/language/sound-problems.md]: I think only the "Unsound implicit downcast" bit has changed.  It's not a static error or warning.

That said, most of your examples are still relevant if the downcast fails at runtime.

File: src/_guides/language/sound-dart.md (hunk 1 above)
Comment (kwalrath) [inline src/_guides/language/sound-dart.md]: I just realized that all these links are to the now-archived dev_compiler project. I've updated them all to be to the dart-lang/sdk location or, in the case of STRONG_MODE.md, to dart-archive/dev_compiler. (Is STRONG_MODE.md still useful?)

File: src/_guides/language/sound-problems.md (hunk 2 above)
Comment (kwalrath) [inline src/_guides/language/sound-problems.md]: Should I remove the entire "Unsound implicit downcast" section? https://kw-www-dartlang-1.firebaseapp.com/guides/language/sound-problems#unsound-implicit-downcast

**Comment by kwalrath** [comment]
PTAL @vsmenon

**Comment by kwalrath** [comment]
I'm going to commit these changes. @vsmenon please let me know if I need to fix anything.

**Pull request merged**